### PR TITLE
Add iosxr_hostname RM

### PR DESCRIPTION
**Pull request by ashwini-mhatre**
##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/iosxr_hostname.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+  type: dict
+  sample: >
+    This output will always be in the same format as the
+    module argspec.
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: when I(state) is C(merged), C(replaced), C(overridden), C(deleted) or C(purged)
+  type: list
+  sample:
+    - sample command 1
```
Comment (NilashishC) [inline plugins/modules/iosxr_hostname.py:260]: Please update this.

File: tests/integration/network-integration.cfg [hunk 2]
```diff
@@ -1,4 +1,18 @@
+[defaults]
+inventory=inventory
+host_key_checking=False
+#gathering=no
+interpreter_python=/Users/amhatre/ansible_venvs/py3.6.10/bin/python
```
Comment (NilashishC) [inline tests/integration/network-integration.cfg:5]: I think we need to revert these changes?

File: tests/integration/test-inventory [hunk 3]
```diff
@@ -0,0 +1,27 @@
+[iosxr]
```
Comment (NilashishC) [inline tests/integration/test-inventory:1]: Same as above.

File: plugins/modules/iosxr_hostname.py (hunk 1 above)
Comment (ashwini-mhatre) [reply plugins/modules/iosxr_hostname.py]: ack

File: tests/integration/test-inventory (hunk 3 above)
Comment (ashwini-mhatre) [reply tests/integration/test-inventory]: ack

**Review by KB-perByte — APPROVED** [review]
